### PR TITLE
24.3 Add config for marking broken integration tests

### PR DESCRIPTION
**Pull request by strtgbb**
### CI Fix or Improvement
Add [tests/integration/broken_tests.json](https://github.com/Altinity/ClickHouse/compare/customizations/24.3.9...strtgbb:ClickHouse:override_broken_tests?expand=1#diff-97496b5b35d32f1aaceccd0acfcba288e6d270339b158de1b0a066a4e6ea0eb3) which is used to selectively replace FAILED and ERROR statuses with BROKEN

Supersedes https://github.com/Altinity/ClickHouse/pull/469

### CODE REVIEW

File: tests/ci/integration_tests_runner.py [hunk 1]
```diff
@@ -464,6 +464,20 @@ def _get_parallel_tests_skip_list(repo_path):
             skip_list_tests = json.load(skip_list_file)
         return list(sorted(skip_list_tests))
 
+    @staticmethod
+    def _get_broken_tests_list(repo_path):
+        skip_list_file_path = f"{repo_path}/tests/integration/broken_tests.json"
```
Comment (Enmk) [inline tests/ci/integration_tests_runner.py:469]: Why `broken_tests.json` but not some kind of `xfailed_test.json` ?

File: tests/ci/integration_tests_runner.py [hunk 2]
```diff
@@ -911,6 +929,13 @@ def run_impl(self, repo_path, build_path):
             group_counters, group_test_times, log_paths = self.try_run_test_group(
                 repo_path, group, tests, MAX_RETRY, NUM_WORKERS
             )
+
+            for fail_status in ("ERROR", "FAILED"):
```
Comment (Enmk) [inline tests/ci/integration_tests_runner.py:933]: It would be really nice to not only match name of the test, but also failure reason/error code/whatever.

**Review by Enmk — CHANGES_REQUESTED** [review]
Please fix mentioned issues.

Also, were you considering usage of `@pytest.mark.xfail`?
First of all: it is kind of standard to pytest, second it allows point-xfail and more....

https://docs.pytest.org/en/stable/how-to/skipping.html#xfail
https://docs.pytest.org/en/stable/reference/reference.html#pytest-mark-xfail-ref

File: tests/ci/integration_tests_runner.py (hunk 1 above)
Comment (strtgbb) [reply tests/ci/integration_tests_runner.py:469]: Naming was chosen based on `run_impl()` understanding `BROKEN`, but not `XFAIL`. If another name for the file makes more sense, I don't have an issue with changing it. We could also add full support for `XFAIL` status if needed.

File: tests/ci/integration_tests_runner.py (hunk 2 above)
Comment (strtgbb) [reply tests/ci/integration_tests_runner.py:933]: I can look into that. Do we want to simply do `test_name:error_message` or try to copy Testflows syntax?

**Comment by strtgbb** [comment]
> Also, were you considering usage of `@pytest.mark.xfail`? First of all: it is kind of standard to pytest, second it allows point-xfail and more....

Yes, pytest marks were considered, however that requires modifying test code and makes it less convenient to see a list of all broken tests.

**Comment by strtgbb** [comment]
I haven't added a comment field yet because the error message seems clear enough. It will be trivial to add such when we require it.

**Comment by vzakaznikov** [comment]
> I haven't added a comment field yet because the error message seems clear enough. It will be trivial to add such when we require it.

No, let's support for a comment field now. We don't want to revisit this functionality for such a small thing.

**Comment by strtgbb** [comment]
> let's add support for a comment field now.

Done. I don't know how to propagate the reason field upwards at this time, so it will just serve to help someone who is looking at the config file.

File: tests/ci/integration_tests_runner.py (hunk 1 above)
Comment (strtgbb) [reply tests/ci/integration_tests_runner.py:469]: Marking as resolved based on meeting last week

**Review by Enmk — APPROVED** [review]
LGTM

**Pull request merged**